Account Recovery: Make `Generate strong password` button functional.

southp opened the pull request:
## Summary
This PR is based on #13364 and makes the "Generate strong password" button functional.
![demo-strong](https://cloud.githubusercontent.com/assets/1842898/25472675/e70e5858-2b5e-11e7-8247-416f7cec0f40.gif)

## Test Plan
1. Open an incognito window and make sure you are logged out.
1. Visit http://calypso.localhost:3000/account-recovery
1. Choose a test account and proceed as the instructions. At the final "New Password" page, click the "Generate strong password" button, and you should see a strong password is generated for you.

a8ci18n commented on client/account-recovery/account-recovery-root/index.jsx:
```diff
@@ -5,43 +5,87 @@ import React from 'react';
+		},
+		[ STEPS.RESET_PASSWORD_CONFIRM ]: {
+			trackerTitle: 'Account Recovery > New Password',
+			documentHeadTitle: translate( 'Reset Password < New Password' ),
```
Hi! I've found a possible matching string that has already been translated **79** times:
`translate( 'Repeat New Password' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/account-recovery/account-recovery-root/index.jsx:
```diff
@@ -5,43 +5,87 @@ import React from 'react';
 		[ STEPS.FORGOT_USERNAME ]: {
 			trackerTitle: 'Account Recovery > Forgot Username',
-			documentHeadTitle: translate( 'Forgot Username ‹ Account Recovery' ),
+			documentHeadTitle: concatHeadTitle( translate( 'Forgot Username' ), translate( 'Account Recovery' ) ),
```
Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Forgot your username?' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

omarjackman commented on client/account-recovery/reset-password-confirm-form/index.jsx:
```diff
@@ -60,6 +68,8 @@ class ResetPasswordConfirmForm extends Component {
 
 		const { newPassword } = this.state;
 
+		const capturePasswordInputRef = ( passwordInput ) => this.passwordInput = passwordInput;
```
Since this isn't the actual password shouldn't this be `const capturePasswordInputRef = ( passwordInputRef ) => this.passwordInputRef = passwordInputRef;`

southp replied:
That's a good idea. Updated in 79e6bb0.

omarjackman commented on client/account-recovery/reset-password-confirm-form/index.jsx:
```diff
@@ -60,6 +68,8 @@ class ResetPasswordConfirmForm extends Component {
 
 		const { newPassword } = this.state;
 
+		const capturePasswordInputRef = ( passwordInput ) => this.passwordInput = passwordInput;
```
Also, can we make this a class method instead. Then we can use `this.capturePasswordInpurRef` below

rodrigoi commented:
@southp do you need help moving this forward?

southp commented:
@rodrigoi That would be great :) I've finally found some time to rebase & addressed the feedbacks, so it's ready for review now.

Of course, @omarjackman your second look will be also much appreciated if you have some bandwidth :)

southp commented:
We are not likely to revisit this project any sooner. I'm closing this to reduce noise for myself.
